### PR TITLE
feat(flags): tags

### LINKED ISSUE #13146: Mark feature flag as temporary or permanent
## Is your feature request related to a problem?

I'm trying to migrate from LaunchDarkly to PostHog.

One feature I'm missing is the ability to mark a feature flag as "temporary" or "permanent", to make cleanup and management of flags much easier.

## Describe the solution you'd like

Ability to Mark flags as temporary when you intend to remove them after releasing a feature. This lets you find and scrub those flags with ease.  

Ability to Mark flags as permanent when keeping them in your code long-term. Use them for operational things like disabling non-essential features during a spike in traffic.

### PR DESCRIPTION
**Pull request by liyiy**
## Problem

Users currently have no way to categorize or label their feature flags
## Changes
Screen Shot 2023-01-11 at 8 57 28 PM
Screen Shot 2023-01-11 at 8 57 34 PM
Screen Shot 2023-01-11 at 8 57 39 PM

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/migrations/0288_add_tags_to_feature_flags.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    operations = [
+        migrations.RemoveConstraint(
+            model_name="taggeditem",
+            name="exactly_one_related_object",
+        ),
+        migrations.AddField(
+            model_name="taggeditem",
+            name="feature_flag",
+            field=models.ForeignKey(
+                blank=True,
```
Comment (liyiy) [inline posthog/migrations/0288_add_tags_to_feature_flags.py:22]: @alexkim205 Getting migration error "field is not null and will lock the table will migrating" even though I'm using blank + null = True, do you have any idea on why this is the case?

Comment (alexkim205) [reply posthog/migrations/0288_add_tags_to_feature_flags.py:22]: I don't see anything wrong with the migration and the error message might be misleading. My guess is that the regex for checking if our migrations are safe is getting caught on this line of the migration, thinking that we're adding a new field here that is not null.

```
CREATE UNIQUE INDEX "unique_feature_flag_tagged_item" ON "posthog_taggeditem" ("tag_id", "feature_flag_id") WHERE "feature_flag_id" IS NOT NULL;
```

We already added this field before with 
```
ALTER TABLE "posthog_taggeditem" ADD COLUMN "feature_flag_id" integer NULL CONSTRAINT 
```

I haven't been able to test it locally but would changing this line (https://github.com/PostHog/posthog/blob/45b4f7092050e606454370cdd019b3b15de23c75/posthog/management/commands/test_migrations_are_safe.py#L17-L17) to:

```
re.findall(r"(?<!DROP|CREATE UNIQUE INDEX) (NOT NULL|DEFAULT)", sql, re.M & re.I)
```

make sure we're not catching these false positives in the test?

Comment (EDsCODE) [reply posthog/migrations/0288_add_tags_to_feature_flags.py:22]: ^^ Migration check is too aggressive

File: posthog/management/commands/test_migrations_are_safe.py [hunk 2]
```diff
@@ -15,6 +15,7 @@ def run_and_check_migration(variable):
                 sql = call_command("sqlmigrate", results[0], results[1])
                 if (
                     re.findall(r"(?<!DROP) (NOT NULL|DEFAULT)", sql, re.M & re.I)
+                    and not re.match(r"(?<!CREATE UNIQUE INDEX) (NOT NULL)", sql)
```
Comment (liyiy) [inline posthog/management/commands/test_migrations_are_safe.py:18]: Was having trouble with the regex of finding all "NOT NULL" instances that aren't followed by CREATE UNIQUE TOGETHER / ALTER TABLE so hoping to get this in and then take it out after the PR is merged 🤗

File: posthog/models/tagged_item.py [hunk 3]
```diff
@@ -61,6 +61,9 @@ class TaggedItem(UUIDModel):
     action: models.ForeignKey = models.ForeignKey(
         "Action", on_delete=models.CASCADE, null=True, blank=True, related_name="tagged_items"
     )
+    feature_flag: models.ForeignKey = models.ForeignKey(
```
Comment (EDsCODE) [inline posthog/models/tagged_item.py:64]: Should add a test of feature_flag in `posthog/models/test/test_tagged_item_model.py`

**Review by EDsCODE — APPROVED** [review]
lgtm! just make sure to add test

**Pull request merged**